src/Microsoft.DotNet.Interactive.Formatting/HtmlFormatter.cs

```diff
@@ -25,6 +68,32 @@ public static class HtmlFormatter
         internal static ITypeFormatter GetDefaultFormatterForAnyEnumerable(Type type) =>
             FormattersForAnyEnumerable.GetFormatter(type, false);
 
+        internal static IHtmlContent FormatObjectAsPlainText(FormatContext context, object value)
+        {
+            using var writer = Formatter.CreateWriter();
+            Formatter.FormatTo(value, context, writer, PlainTextFormatter.MimeType);
+            var text = writer.ToString();
+            return FormatString(text);
+        }
+
+        internal static IHtmlContent FormatString(string text)
+        {
+            var html = text.HtmlEncode();
+
+            if (PlainTextPreformat)
```

Comment:
I initially misunderstood this name to mean "something we do before formatting". 

Looking more closely at this and the associated properties, I think it might be more flexible and easier to understand by removing them and doing something like:

```csharp
public Func<Tag> PlainTextWrapperTag> { get; set; } = () => new Tag("div");
```

This lets people control all of these style elements and add others.

Reply:
I do see the problem.

Another approach would be to emit consistent and documented CSS classes, e.g. emit any and all of `plaintext`, `dotnet-plaintex`, `fsharp-plaintext`, `csharp-plaintext`, ... plus tags for other outputs - e.g. `dotnet-object` etc.

Then have a helper function `CSS`:

    CSS ".fsharp-plaintext { text-align: left; white-space: pre; }"

which applies the formatting selectively.  Some advantages being it is general and requires almost no documentation (beyond documenting the tags emitted).   

So in this case we would always emit `<div class="dotnet-plaintext">...</div>`....  And perhaps also a general wrapper `<div class="dotnet-fsharp">...</div>` at the outer of all F# kernel outputs (likewise C#, powershell) to allow them to be formatted selectively.

Reply:
I think ultimately we'll want to emit documented, stable CSS classes like this. I'd like to design it independent of this PR. In the meantime, I'm concerned that creating fine-grained abstractions for this in .NET (a) is necessarily limited and (b) will later clash with the more natural CSS-class approach.